lib/libc/newlib/libc-hooks.c

```diff
@@ -450,7 +450,21 @@ void __retarget_lock_release_recursive(_LOCK_T lock)
 	__ASSERT_NO_MSG(lock != NULL);
 	k_mutex_unlock((struct k_mutex *)lock);
 }
-#endif /* CONFIG_MULTITHREADING */
+#else /* CONFIG_MULTITHREADING && _RETARGETABLE_LOCKING */
```

Comment:
Do you want to look into sharing the common malloc implementation instead of using the newlib one? I realize it's not directly aligned with your goals here, but I don't know anyone else poking at newlib code...

Reply:
I cannot find a definite way to exclude a function from a static library during linking. IIRC, providing multiple implementation may result in the linker either complaining, or picking one depending on linking order. If it is latter, cmake scripts may need to be changed to ensure ordering all the time.